runners/flink/src/test/java/org/apache/beam/runners/flink/metrics/FlinkMetricContainerTest.java

```diff
@@ -106,6 +110,39 @@ public void testGauge() {
     assertThat(flinkGauge.getValue().getValue(), is(42L));
   }
 
+  @Test
+  public void testMonitoringInfoUpdate() {
+    FlinkMetricContainer container = new FlinkMetricContainer(runtimeContext);
+    MetricsContainer step = container.getMetricsContainer("step");
+
+    SimpleCounter userCounter = new SimpleCounter();
+    when(metricGroup.counter("ns1.metric1")).thenReturn(userCounter);
+
+    SimpleCounter elemCounter = new SimpleCounter();
+    when(metricGroup.counter("beam.metric:element_count:v1")).thenReturn(elemCounter);
```

Comment:
The current namespace parsing logic [above](https://github.com/apache/beam/pull/7408/files#diff-bd906385f3e3ff74094985af80da5a41R109) just takes the everything after the first `:` as the "name", so this name is `metric:element_count:v1`.

I don't see any reason that having additional `:`s in the "name" is a problem, but I saw [a note about it (only related to user metrics, though) in SimpleMonitoringInfoBuilder](https://github.com/apache/beam/blob/5716dba6d10f32dfa8ab807ffacc2e68f9267225/runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/SimpleMonitoringInfoBuilder.java#L114-L115), so open to whether there's a better way to think about it here.